client/src/components/RelatedObjectNoteModal.js

```diff
@@ -44,7 +45,8 @@ const RelatedObjectNoteModal = ({
           submitForm
         }) => {
           const isJson = note.type !== NOTE_TYPE.FREE_TEXT
-          const jsonFields = isJson && note.text ? JSON.parse(note.text) : {}
+          const jsonFields =
+            isJson && note.text ? utils.parseJsonSafe(note.text) : {}
```

Comment:
Actually this could also just be written as const jsonFields = utils.parseJsonSafe(note.text), ending with the same result, but maybe it's good to keep the conditional for better understanding?

Reply:
You don't want to parse the note text if the note is definitely not JSON (waste of CPU cycles). And someone may have actually typed valid JSON in a free text note (unlikely, put possible).